### DISCUSSION TITLE
RLS with Edge Functions

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#8080] [Questions]
Supabase uses Row Level Security thing for security layer over data tables, And I can set auth for signed in user in my flutter app, using setAuth function of supabase flutter client, but how can I sync the auth state of the signed user in and on the supabase cloud for edge function ? or should I use an API key that can bypass RLS in edge functions?

If you try to perform operations on edge function, with RLS enabled it will not work, as for supabase client lying in supabase cloud (edge function) there is no user signed. 

What can be the optimal way to keep the RLS and use edge functions?

**Comment by GaryAustin1** [comment]
You can certainly use service_role to bypass RLS and manage access in your edge function.

Or if the function is called from the browser/app using the supabase client there, the jwt is passed to the function.
This example from SB shows using the jwt to an RLS table.

https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts
